internal/mode/static/telemetry/collector.go

```diff
@@ -0,0 +1,78 @@
+package telemetry
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/state"
+)
+
+type Collector interface {
+	CollectData() Data
+}
+
+type DataCollector struct {
+	ctx       context.Context
+	k8sClient client.Client
+	processor state.ChangeProcessor
+}
+
+type GraphResourceCount struct {
+	Gateways           int
+	GatewayClasses     int
+	HTTPRoutes         int
+	ReferencedSecrets  int
+	ReferencedServices int
+}
+
+func NewDataCollector(
+	ctx context.Context,
+	k8sClient client.Client,
+	processor state.ChangeProcessor,
+) *DataCollector {
+	return &DataCollector{
+		ctx:       ctx,
+		k8sClient: k8sClient,
+		processor: processor,
+	}
+}
+
+func (c DataCollector) CollectData() Data {
+	nodeCount := collectNodeCount(c.ctx, c.k8sClient)
+	graphResourceCount := collectGraphResourceCount(c.processor)
+
+	data := Data{
+		NodeCount:          nodeCount,
+		GraphResourceCount: graphResourceCount,
+	}
+
+	return data
+}
+
+func collectNodeCount(ctx context.Context, k8sClient client.Client) int {
```

Comment:
Both `collectNodeCount` and `collectGraphResourceCount` are still WIP and will have unit tests added.